palace/drivers/basesolver.hpp

```diff
@@ -68,12 +67,22 @@ class BaseSolver
                          double time) const;
   void PostprocessFields(const PostOperator &postop, int step, double time) const;
 
+  // Postprocess granular error indicator to file. The argument normalized indicates if
+  // supplied indicators have already been normalized.
+  void PostprocessErrorIndicators(const std::string &name, int step, double time,
+                                  const ErrorIndicators &indicators, bool normalized) const;
+  // Write a string labeled error indicator. Used for writing statistics.
+  void PostprocessErrorIndicators(const std::string &name,
+                                  const ErrorIndicators &indicators) const;
+
 public:
   BaseSolver(const IoData &iodata, bool root, int size = 0, int num_thread = 0,
              const char *git_tag = nullptr);
   virtual ~BaseSolver() = default;
 
-  virtual void Solve(std::vector<std::unique_ptr<mfem::ParMesh>> &mesh) const = 0;
+  // Performs a solve using the mesh sequence, and report error indicators.
+  virtual ErrorIndicators
```

Comment:
I'd rather not, as there's no possibility of this ever being a null state, so pointers adds extra verbosity without a bought feature. It even suggests that any call to `Solve` should be checking for null status before accessing the variable. Returning by value guarantees that the variable is a valid return. `ErrorIndicators` also has move semantics, so even without NRVO, which would almost certainly trigger, this would end up being equivalent to a pointer return.